File: Codes/2239-find-closest-number-to-zero.cpp
/*
 * @Author: LetMeFly
 * @Date: 2025-01-20 13:25:25
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-01-20 13:29:40
 */
#ifdef _WIN32
#include "_[1,2]toVector.h"
#endif

class Solution {
public:
    int findClosestNumber(vector<int>& nums) {
        int ans = nums[0];
        for (int t : nums) {
            if (abs(t) < abs(ans) || abs(t) == abs(ans) && ans < 0) {
                ans = t;
            }
        }
        return ans;
    }
};

File: Codes/2239-find-closest-number-to-zero.py
'''
Author: LetMeFly
Date: 2025-01-20 13:31:49
LastEditors: LetMeFly.xyz
LastEditTime: 2025-01-20 13:32:00
'''
from typing import List

class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        ans = nums[0]
        for t in nums:
            if abs(t) < abs(ans) or abs(t) == abs(ans) and ans < 0:
                ans = t
        return ans